### PR TITLE
Update syncd stop script to collect saisdkdump during orch abort

### PR DESCRIPTION
**Pull request by vivekrnv**
Signed-off-by: Vivek Reddy Karri <vkarri@nvidia.com>

#### Why I did it

- Update the syncd stop flow to collect saisdkdump and save it to /var/log/orch_abrt_status/ location during a SAI programming failure by orchagent. 
   1) Limit the number of such archived to 10
   2) Create a tmp file as a sync mechanism with auto-techsupport process
- Update the swss stop flow to remove PortInitDone. This is done so as to provide other daemons a reliable way of checking if switch_init is successful after services are deemed started.  (Eg: Before running saisdkdump, switch should be initialized. techsupport will use this info to determine that)

#### How to verify it

```

root@r-bulldog-03:/home/admin# sonic-db-cli STATE_DB SET ORCH_ABRT_STATUS 1

root@r-bulldog-03:/home/admin# systemctl restart swss

Oct 28 21:10:52.892667 r-bulldog-03 INFO syncd.sh[26913]: sai_sdk_dump_1666991448.tar.gz collected before taking stopping syncd

root@r-bulldog-03:/home/admin# tar -tvf /var/log/orch_abrt_status/sai_sdk_dump_1666991448.tar.gz
drwxr-xr-x root/root         0 2022-10-28 21:10 ./
-rw-r--r-- root/root   4883498 2022-10-28 21:10 ./sdkdump_ext_cr_001-28_10_2022-21_10_50-980950.udmp
-rw-r--r-- root/root   3964382 2022-10-28 21:10 ./sai_sdk_dump_10_28_2022_09_10_PM
-rw-r--r-- root/root       288 2022-10-28 21:10 ./sdkdump_ext_meta_001-28_10_2022-21_10_50-407583
-rw-r--r-- root/root       288 2022-10-28 21:10 ./sdkdump_ext_meta_001-28_10_2022-21_10_50-980950
-rw-r--r-- root/root   2656360 2022-10-28 21:10 ./sai_sdk_dump_10_28_2022_09_10_PM.json
-rw-r--r-- root/root   4883498 2022-10-28 21:10 ./sdkdump_ext_cr_001-28_10_2022-21_10_50-407583.udmp
-rw-r--r-- root/root   4883498 2022-10-28 21:10 ./sdkdump_ext_cr_001-28_10_2022-21_10_51-573167.udmp
-rw-r--r-- root/root       288 2022-10-28 21:10 ./sdkdump_ext_meta_001-28_10_2022-21_10_51-573167
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: files/scripts/syncd.sh [hunk 1]
```diff
@@ -75,6 +98,17 @@ function waitplatform() {
 
 function stopplatform1() {
 
+    if [[ x"$(${SONIC_DB_CLI} STATE_DB GET ORCH_ABRT_STATUS)" == x"1" ]]; then
+        # Collecting saisdkdump before restarting syncd
+        # Runs when orchagent is aborted because of SAI failure.
+        # Only enabled for mellanox platform
+        if [[ x$sonic_asic_platform == x"mellanox" ]]; then
+            collect_saisdkdump
```
Comment (dgsudharsan) [inline files/scripts/syncd.sh:106]: Can you run this with timeout? In case if saisdkdump is stuck it means syncd restart would also be stuck

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: files/scripts/syncd.sh [hunk 2]
```diff
@@ -75,6 +98,17 @@ function waitplatform() {
...
+        fi
+        # This is used to notify auto-techsupport process
+        touch /tmp/saidump_collection_notify_flag
```
Comment (dgsudharsan) [inline files/scripts/syncd.sh:109]: Should we touch only when auto tech support is enabled?

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: files/scripts/syncd.sh (hunk 1 above)
Comment (vivekrnv) [reply files/scripts/syncd.sh:106]: Good catch, will update

File: files/scripts/syncd.sh (hunk 2 above)
Comment (vivekrnv) [reply files/scripts/syncd.sh:109]: Actually no, auto-TS is equipped to delete this file even if the feature is disabled. Besides, It's not clean and rather clumsy to use AUTO_TS config checks in this script

File: files/scripts/syncd.sh (hunk 1 above)
Comment (vivekrnv) [reply files/scripts/syncd.sh:106]: Updated

File: files/scripts/syncd.sh [hunk 3]
```diff
@@ -11,7 +11,7 @@ function collect_saisdkdump() {
     /usr/bin/docker exec syncd$DEV rm -rf ${TMP_DMP_DIR}
     /usr/bin/docker exec syncd$DEV rm -f /tmp/${sai_dump_filename_epoch}.tar
     /usr/bin/docker exec syncd$DEV mkdir -p ${TMP_DMP_DIR}
-    /usr/bin/docker exec syncd$DEV saisdkdump -f ${TMP_DMP_DIR}/${sai_dump_filename} > /dev/null
+    timeout 1m bash -c "/usr/bin/docker exec syncd$DEV saisdkdump -f ${TMP_DMP_DIR}/${sai_dump_filename} > /dev/null"
```
Comment (dgsudharsan) [inline files/scripts/syncd.sh:14]: 1 minute seems to be a huge timeout. Can you check how long saisdkdump takes in each platform and come up with 2x of max time taken. Ideally saisdkdump should finish within 5-6 seconds. If that's the scenario we can add timeout of 15 seconds

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: files/scripts/syncd.sh (hunk 3 above)
Comment (vivekrnv) [reply files/scripts/syncd.sh:14]: This is worst case scenario right. And we can't predict how long will the saisdkdump of other vendors would take. Anyways, i'll set it to 30s. Seems reasonable?

**Review by dgsudharsan — APPROVED** [review]

**Pull request closed without merging**